sktime/forecasting/arch/_uarch.py

```diff
@@ -0,0 +1,362 @@
+        models support all methods.
+        In particular, volatility models that do not evolve in squares such as
+        EGARCH or TARCH
+        do not support the 'analytic' method for horizons > 1.
+    simulations : int
+        Number of simulations to run when computing the forecast using either
+        simulation or bootstrap.
+    rng : callable, optional
+        Custom random number generator to use in simulation-based forecasts.
+        Must produce random samples using the syntax rng(size) where size the
+        2-element tuple (simulations, horizon).
+    random_state : RandomState, optional
+        NumPy RandomState instance to use when method is 'bootstrap'
+    reindex : bool, optional
+        Whether to reindex the forecasts to have the same dimension as the series
+        being forecast. Prior to 4.18 this was the default. As of 4.19 this is now
+        optional. If not provided, a warning is raised about the future change in
+        the default which will occur after September 2021.
+
+    See Also
+    --------
+    StatsForecastARCH
+    StatsForecastGARCH
+
+    References
+    ----------
+    .. [1] Jason Brownlee. How to Model Volatility with ARCH and GARCH for Time Series
+       Forecasting.
+       https://machinelearningmastery.com/develop-arch-and-garch-models-for-time-series-
+       forecasting-in-python/
+    .. [2] GitHub repository of arch package (soft dependency).
+       https://github.com/bashtage/arch
+    .. [3] Documentation of arch package (soft dependency). Forecasting Volatility with
+       ARCH and it's variants.
+       https://arch.readthedocs.io/en/latest/univariate/introduction.html
+
+    Examples
+    --------
+    >>> from sktime.datasets import load_airline
+    >>> from sktime.forecasting.garch import ARCH
+    >>> y = load_airline()
+    >>> forecaster = ARCH()
+    >>> forecaster.fit(y)
+    ARCH(...)
+    >>> y_pred = forecaster.predict(fh=1)
+    """
+
+    _tags = {
+        "scitype:y": "univariate",
+        "y_inner_mtype": "pd.Series",
+        "X_inner_mtype": "pd.DataFrame",
+        "requires-fh-in-fit": False,
+        "handles-missing-data": False,
+        "python_dependencies": "arch",
+        "ignores-exogeneous-X": True,
+    }
+
+    def __init__(
+        self,
+        mean="Constant",
+        lags=0,
+        vol="GARCH",
+        p=1,
+        o=0,
+        q=1,
+        power=2.0,
+        dist="Normal",
+        hold_back=None,
+        rescale=False,
+        update_freq=0,
+        disp=False,
+        starting_values=None,
+        cov_type="robust",
+        show_warning=False,
+        first_obs=None,
+        last_obs=None,
+        tol=None,
+        options=None,
+        backcast=None,
+        params=None,
+        start=None,
+        align="origin",
+        method="simulation",
+        simulations=10,
+        rng=None,
+        random_state=None,
+        reindex=False,
+    ):
+        _check_soft_dependencies("arch", severity="error")
```

Comment:
I'm a bit unsure if this is required, or presence of the python_dependencies tag is enough. @fkiraly can you please comment?

Reply:
I think it is not required; it was according to old extension template.